README.md

```diff
@@ -0,0 +1,88 @@
+# bls-wallet
+
+An Ethereum Layer 2 smart contract wallet that uses [BLS signatures](https://en.wikipedia.org/wiki/BLS_digital_signature) and aggregated transactions to reduce gas costs.
+
+## Components
+
+See each component's directory `README` for more details.
+
+### Aggregator
+
+Service which aggregates BLS wallet transactions.
+
+### Clients
+
+TS/JS Client libraries for web apps and services.
+
+### Contracts
+
+`bls-wallet` Solidity contracts.
+
+### Extension
+
+Quill browser extension used to manager BLS Wallets and sign transactions.
+
+### Signer
+
+TS/JS BLS Signing lib.
+
+## Dependencies
+
+### Required
+
+- [NodeJS](https://nodejs.org)
+- [Yarn](https://yarnpkg.com/getting-started/install) (`npm install -g yarn`)
+- [Deno](https://deno.land/#installation)
+
+### Optional (Recomended)
+
+- [nvm](https://github.com/nvm-sh/nvm#installing-and-updating)
+- [docker-compose](https://docs.docker.com/compose/install/)
+- [MetaMask](https://metamask.io/)
+
+## Setup
+
+Run the repo setup script
+```sh
+./setup.ts
+```
+
+### Local
+
+TODO, it's a wee bit complicated. Wait on `create2` contract deploys.
+
+### Arbitrum Testnet (Rinkeby Arbitrum Testnet)
+
+You will need two ETH addresses with Rinkeby ETH and their private keys (PK0 & PK1) for running the aggregator. It is NOT recommended that you use any primary wallets with ETH Mainnet assets.
+
+You can get Rinkeby ETH at https://app.mycrypto.com/faucet, and transfer it into the Arbitrum testnet via https://bridge.arbitrum.io/. Make sure when doing so that your network is set to Rinkeby in MetaMask.
+
+#### Env Updates
+
+`./aggregator/.env`
+```
+PRIVATE_KEY_AGG=PK0
+PRIVATE_KEY_ADMIN=PK1
+```
+
+From `./contracts/networks/rinkarby.json` (JSON key path):
+
+`./extension/.env`
+```
+VERIFICATION_GATEWAY_ADDRESS=addresses.verificationGateway
+```
+
+## Run
+
+```sh
+docker-composer up -d postgres # Or see local postgres instructions in ./aggregator/README.md#PostgreSQL
+cd ./aggregator
+./programs/aggregator.ts
+```
+
+In a seperate terminal/shell instance
+```sh
+cd ./extension
+yarn run dev:chrome # or dev:firefox, dev:opera
+```
+Then install `./extension/{browser}/manifest.json` in your browser of choice.
```

Comment:
Let's add some more detail / link to guide.

E.g. in chrome, you need to drag the `./extension/chrome` *folder* into the extensions page, giving it the manifest.json file doesn't work.